tests/tests/test_adacore.py

```diff
@@ -103,6 +104,10 @@ def set_up(self):
     )
     assert message in testsuite_logs(caplog)
 
+    # Check that XPASS prints the .opt XFAIL message
+    message = "XPASS           xpass-msg: testing XPASS message"
+    assert message in testsuite_logs(caplog)
```

Comment:
The change of behavior occur in `ClassicTestDriver`: could you move this test to `test_classic.py`? Also I would suggest to check the test message using the report index: this API is public and supposed to be stable, while the format of logs could change in the future. Something like:

```python
assert suite.report_index.entries["xpass-msg"].msg == "testing XPASS message”
```

Thanks for the contribution!